### PR TITLE
Improve chatbot accessibility and UX

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

This PR includes:
- improved HTML semantics specific to chatbots
- improved mobile UX
- new like/dislike icons - like + dislike is also filled when clicked
- allow switching between chatbot designs using `bubble` or `panel` via a new `layout` param

Screenshot 2023-09-27 at 14 29 26

Screenshot 2023-09-27 at 16 44 32

Screenshot 2023-09-27 at 17 55 05
Screenshot 2023-09-27 at 17 54 08

demo: 

```
import gradio as gr

def test(x: gr.LikeData):
    print(x.index, x.value, x.liked)

with gr.Blocks() as demo:
    c = gr.Chatbot([["Donec id justo. Proin sapien ipsum, porta a, auctor quis, euismod ut, mi.", "Pellentesque libero tortor, tincidunt et, tincidunt eget, semper nec, quam. Aenean viverra rhoncus pede. Etiam sit amet orci eget eros faucibus tincidunt. In turpis.. Etiam iaculis nunc ac metus."], ["Phasellus viverra nulla ut metus varius laoreet. Duis lobortis massa imperdiet quam.", "Suspendisse potenti. Curabitur nisi. Sed lectus. Quisque malesuada placerat nisl. Vestibulum eu odio. Donec vitae orci sed dolor rutrum auctor. Fusce neque. Curabitur ligula sapien, tincidunt non, euismod vitae, posuere imperdiet, leo. Suspendisse feugiat."]], show_copy_button=True, layout="bubble")
    c.like(test, None, None)
    
demo.launch()
```

Closes: #5646 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Review by dawoodkhan82 — CHANGES_REQUESTED** [review]
Seems like the avatar image position changed

Screenshot 2023-09-29 at 10 52 43 AM

File: js/chatbot/static/ChatBot.svelte [hunk 1]
```diff
@@ -259,52 +284,85 @@
...
-		align-self: flex-end;
+		align-self: flex-start;
 		border-bottom-right-radius: 0;
+		text-align: right;
 	}
 	.bot {
 		border-bottom-left-radius: 0;
-		padding-left: var(--spacing-xxl);
+		text-align: left;
 	}
```
[the remark this answers is not in the document]
Comment (dawoodkhan82) [reply js/chatbot/static/ChatBot.svelte:339]: Let's leave it the way it was before, and maybe we can add a param to customize it if devs request.

**Comment by dawoodkhan82** [comment]
The panel layout is clean! The copy button position seems a bit awkward to me tho.

Screenshot 2023-09-29 at 10 56 28 AM

**Comment by dawoodkhan82** [comment]
The new button layouts for like and copy are super nice!

**Comment by dawoodkhan82** [comment]
Just tested everything, and it all looks and works great! One small nit is keeping the thumbs up/down button in a "clicked" UI state. So users know which messages they've liked when scrolling through the history.

**Comment by Keldos-Li** [comment]
Wait, why use a `<button>` tag to wrap a message...? This doesn't align with the tag semantics at all😰! This is a huge disaster for downstream developers who need to adapt... 😢

I just don't understand... I think ordinary `<div>` can also contain the message information as well...?

**Comment by davies-w** [comment]
How do I use this with ChatInterface?

**Comment by mardelden** [comment]
@hannahblair , looks like someone broke your layout='panel' enhancement in 4.38.0 by introducing bubbles to the panel layout.

**Pull request merged**